SpiceSharp/Simulations/IntegrationMethods/Spice/SpiceIntegrationMethod.cs

```diff
@@ -0,0 +1,341 @@
+
+            // The first point will be after t=0
+            Break = true;
+
+            // Assume a circuit in DC
+            foreach (var state in IntegrationStates)
+                state.Delta = MaxStep;
+        }
+
+        /// <summary>
+        /// Accept the current time point as a final solution
+        /// </summary>
+        /// <param name="simulation">The simulation</param>
+        public override void Accept(TimeSimulation simulation)
+        {
+            // Clear breakpoints
+            while (Time > Breakpoints.First)
+                Breakpoints.ClearBreakpoint();
+            Break = false;
+
+            base.Accept(simulation);
+        }
+
+        /// <summary>
+        /// Continue
+        /// </summary>
+        /// <param name="simulation">Simulation</param>
+        /// <param name="delta">Delta</param>
+        public override void Continue(TimeSimulation simulation, ref double delta)
+        {
+            // Modify the timestep
+            delta = Math.Min(delta, MaxStep);
+
+            // Handle breakpoints
+            if (Time.Equals(Breakpoints.First) || Breakpoints.First - Time <= MinStep)
+            {
+                // Cut integration order
+                Order = 1;
+
+                // Limit the next timestep
+                var mt = Math.Min(_saveDelta, Breakpoints.Delta);
+                delta = Math.Min(delta, 0.1 * mt);
+
+                // Spice will divide the first timestep by 10
+                if (BaseTime.Equals(0.0))
+                    delta /= 10.0;
+
+                // Don't go below MinStep without reason
+                delta = Math.Max(delta, 2.0 * MinStep);
+            }
+            else if (Time + delta >= Breakpoints.First)
+            {
+                Break = true;
+                _saveDelta = delta;
+                delta = Breakpoints.First - Time;
+            }
+
+            base.Continue(simulation, ref delta);
+        }
+
+        /// <summary>
+        /// Probe a new time point
+        /// </summary>
+        /// <param name="simulation">Simulation</param>
+        /// <param name="delta">Timestep</param>
+        public override void Probe(TimeSimulation simulation, double delta)
+        {
+            base.Probe(simulation, delta);
+
+            ComputeCoefficients();
+            Predict(simulation);
+
+            // Save the current timestep
+            _oldDelta = IntegrationStates[0].Delta;
+        }
+
+        /// <summary>
+        /// Indicate that the solution did not converge
+        /// </summary>
+        /// <param name="simulation">Simulation</param>
+        /// <param name="newDelta">The new timestep</param>
+        public override void NonConvergence(TimeSimulation simulation, out double newDelta)
+        {
+            base.NonConvergence(simulation, out newDelta);
+
+            // Limit the timestep and cut the order
+            newDelta = Math.Min(newDelta, IntegrationStates[0].Delta / 8.0);
```

Comment:
@svenboulanger  I would welcome a const for '0'. What is IntegrationStates[0] ?

Reply:
0 is the number of points to go back in time. So 0 is the current time point.